tests/console/validate_zfcp.pm

```diff
@@ -0,0 +1,199 @@
+
+            # Validate scsi devices with specific command for zfcp: lszdev
+            my $names = $lun->{names};
+            $bus_wwpn = "$fcp_channel:$wwpn";
+            validate_script_output("lszdev --no-headings zfcp-lun | grep '$names'",
+                qr/$bus_wwpn/);
+
+            # Set SCSI devices offline and set back online (other states are also possible)
+            my $state_file = "\"/sys/bus/scsi/devices/$hctl/state\"";
+            validate_script_output("cat $state_file", qr/running/);
+            assert_script_run("echo offline > $state_file",
+                fail_message => "state offline could not be set for SCSI device $hctl");
+            validate_script_output("cat $state_file", qr/offline/);
+            assert_script_run("echo running > $state_file",
+                fail_message => "state online could not be set for SCSI device $hctl");
+            validate_script_output("cat $state_file", qr/running/);
+        }
+    }
+}
+
+# The zfcp_ping and zfcp_show commands can probe ports and retrieve information about ports
+# in the attached storage servers and in interconnect elements such as switches, bridges, and hubs.
+# Because the commands are processed by the SAN management server, information can be obtained
+# about ports and interconnect elements that are not connected to your FCP channel.
+# Thus, zfcp_ping and zfcp_show can help to identify configuration problems in a SAN.
+# The zfcp_show command retrieves information about the SAN topology and details about the SAN components.
+sub investigate_san_fabric {
+    my ($fcp_devices) = shift;
+
+    zypper_call("in libzfcphbaapi0");
+    for my $fcp_device (@{$fcp_devices}) {
+        for my $lun (@{$fcp_devices->{fcp_luns}}) {
+            my $wwpn = $lun->{wwpn};
+            script_run("zfcp_ping -t97 $wwpn >> /tmp/zfcp_ping_wwpns.log");
+        }
+    }
+    upload_logs('/tmp/zfcp_ping_wwpns.log');
+    script_run('zfcp_show -v > /tmp/zfcp_show.log');
+    upload_logs('/tmp/zfcp_show.log');
+}
+
+sub run {
+    select_console 'root-console';
+
+    # FCP devices online
+    record_info('FCP devices', 'Verification of FCP devices online');
+    my $fcp_devices = get_fcp_devices_online();
+
+    # Verification of FCP devices
+    record_info('FCP devices', 'Verification of FCP devices');
+    check_fcp_devices_against_installed_system($fcp_devices);
+    verify_fcp_devices_against_infrastructure($fcp_devices);
+    verify_fcp_attributes($fcp_devices);
+    verify_fcp_devices_can_be_set_online_offline($fcp_devices);
+
+    # Verification of LUNs visible as SCSI devices
+    record_info('LUNs SCSI', 'Verification of LUNs visible as SCSI devices');
+    verify_scsi_devices($fcp_devices);
+}
+
+sub post_fail_hook {
+    my ($self) = shift;
+
+    my $fcp_devices_online = get_fcp_devices_online();
+    investigate_san_fabric($fcp_devices_online);
+    $self->SUPER::post_fail_hook();
```

Comment:
super at the top

Reply:
Mainly, in most of the languages the super goes first with respect to inheritance. In perl it does not matter so much and you can dispatch the control to the superclass. I am not very good with the technical details in perl but the SUPER does some deep magic behind. Anyway i see two reasons despite the previous comments. first the post_fail_hook in super class deals with the selection of the console. secondly, we had an issue with the hierarchy of the `$self->SUPER::post_fail_hook`. I found this comment for you https://progress.opensuse.org/issues/46979#note-4

Reply:
I doesn't matter the language, the super goes first in OOP, but that apply to constructors, destructors, this is some sort of 'TearDown' method when there is a fail, tester might want to have some action in different order, it is not like you are cleaning resources on memory, for instance.
About those two reasons, the test use the root console, so it is not going to change to any other one when failing (I guess?). About second reason I don't know status or old issues but seems that you are with me in this, that the whole hierarchy could fail and you don't get this specific info.
I will wait for other reasons here from you or someone else, leaving it open (I might change it at the end :) as strictly is better as you suggested but I don't see much difference for now).